### PR TITLE
Add audio effects to the weapons

### PR DESCRIPTION
**Pull request by Dennisbonke**
This PR adds weapon attack sounds and extends the weapon object to hold an audio clip that should be played on attacking.

OPA-335-336-337

### CODE REVIEW

File: Assets/Scripts/Player/UnitGridCombat.cs [hunk 1]
```diff
@@ -178,6 +182,10 @@ private enum Direction {
 				return;
 			}
 			DetermineAttackAnimation(unitGridCombat);
+			audioSource_.clip = weapon.GetAttackSound();
+			if(audioSource_.clip != null) {
```
Comment (Westlanderz) [inline Assets/Scripts/Player/UnitGridCombat.cs:186]: Just one question, when is the `audioSource_.clip` ever set to `null`, I get the check but just wondering when this case comes up.

**Review by Westlanderz — APPROVED** [review]
Code looks good to me.

**Review by Westlanderz — APPROVED** [review]
LGTM

File: Assets/Scripts/Player/UnitGridCombat.cs (hunk 1 above)
Comment (Dennisbonke) [reply Assets/Scripts/Player/UnitGridCombat.cs:188]: When we don't set any audio for the weapon at all, so yeah this can be null.

**Pull request merged**